components/movementsensor/gpsrtkpmtk/gpsrtkpmtk_test.go

```diff
@@ -0,0 +1,328 @@
+		_, err = cfg.Validate(path)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("invalid ntrip url", func(t *testing.T) {
+		cfg := Config{
+			NtripURL:             "",
+			NtripConnectAttempts: 10,
+			NtripPass:            "somepass",
+			NtripUser:            "someuser",
+			NtripMountpoint:      "NYC",
+			Board:                testBoardName,
+			I2CBus:               testBusName,
+			I2CAddr:              testi2cAddr,
+		}
+		err := cfg.validateNtrip(path)
+		test.That(t, err, test.ShouldBeError,
+			utils.NewConfigValidationFieldRequiredError(path, "ntrip_url"))
+		err = cfg.validateI2C(path)
+		test.That(t, err, test.ShouldBeNil)
+
+		_, err = cfg.Validate(path)
+		test.That(t, err, test.ShouldBeError,
+			utils.NewConfigValidationFieldRequiredError(path, "ntrip_url"))
+	})
+
+	t.Run("invalid i2c bus", func(t *testing.T) {
+		cfg := Config{
+			I2CBus:               "",
+			NtripURL:             "http//fakeurl",
+			NtripConnectAttempts: 10,
+			NtripPass:            "somepass",
+			NtripUser:            "someuser",
+			NtripMountpoint:      "NYC",
+			Board:                testBoardName,
+			I2CAddr:              testi2cAddr,
+		}
+		err := cfg.validateNtrip(path)
+		test.That(t, err, test.ShouldBeNil)
+		err = cfg.validateI2C(path)
+		test.That(t, err, test.ShouldBeError,
+			utils.NewConfigValidationFieldRequiredError(path, "i2c_bus"))
+		_, err = cfg.Validate(path)
+		test.That(t, err, test.ShouldBeError,
+			utils.NewConfigValidationFieldRequiredError(path, "i2c_bus"))
+	})
+
+	t.Run("invalid i2c addr", func(t *testing.T) {
+		cfg := Config{
+			I2CAddr:              0,
+			NtripURL:             "http//fakeurl",
+			NtripConnectAttempts: 10,
+			NtripPass:            "somepass",
+			NtripUser:            "someuser",
+			NtripMountpoint:      "NYC",
+			Board:                testBoardName,
+			I2CBus:               testBusName,
+		}
+		err := cfg.validateNtrip(path)
+		test.That(t, err, test.ShouldBeNil)
+		err = cfg.validateI2C(path)
+		test.That(t, err, test.ShouldBeError,
+			utils.NewConfigValidationFieldRequiredError(path, "i2c_addr"))
+		_, err = cfg.Validate(path)
+		test.That(t, err, test.ShouldBeError,
+			utils.NewConfigValidationFieldRequiredError(path, "i2c_addr"))
+	})
+}
+
+func TestConnect(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	ctx := context.Background()
+	cancelCtx, cancelFunc := context.WithCancel(ctx)
+	g := rtkI2C{
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+		logger:     logger,
+	}
+
+	url := "http://fakeurl"
+	username := "user"
+	password := "pwd"
+
+	// create new ntrip client and connect
+	err := g.connect("invalidurl", username, password, 10)
+	test.That(t, err.Error(), test.ShouldContainSubstring, `Can't connect to NTRIP caster`)
+
+	g.ntripClient = makeMockNtripClient()
+
+	err = g.connect(url, username, password, 10)
+	test.That(t, err, test.ShouldBeNil)
+
+	err = g.getStream("", 10)
+	test.That(t, err.Error(), test.ShouldContainSubstring, `lookup fakeurl`)
```

Comment:
since this function is for testing connect I think there should be a separate `testGetStream` function. also not sure if its worth testing for this random error so if there's not a good way to actually test getting the stream maybe just remove.

Reply:
It's really not testing anything. I'll remove it